### PR TITLE
[Media API]: Requesting API Review

### PR DESCRIPTION
**Pull request by dianaqu**

### CODE REVIEW

File: specs/Media.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2StagingIsAudioPlayingChangedEventHandler;
+
+[uuid(71c906d9-4a4d-4dbe-aa1b-db64f4de594e), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
```
Comment (david-risney) [inline specs/Media.md:118]: This should probably be a bigger number
```suggestion
interface ICoreWebView2_6 : ICoreWebView2 {
```

File: specs/Media.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  /// if there are audio currently playing.
+  ///
+  /// \snippet AudioComponent.cpp IsAudioPlaying
+  [propget] HRESULT IsAudioPlaying([out, retval] BOOL* value);
+}
+
+/// Implements the interface to receive `IsAudioPlayingChanged` events.  Use the
+/// IsAudioPlaying method to get the audible state.
+[uuid(5DEF109A-2F4B-49FA-B7F6-11C39E513328), object, pointer_default(unique)]
+interface ICoreWebView2StagingIsAudioPlayingChangedEventHandler : IUnknown {
```
Comment (david-risney) [inline specs/Media.md:157]: Please use approximate final names - remove staging from here and also from above in the sample code

```suggestion
interface ICoreWebView2IsAudioPlayingChangedEventHandler : IUnknown {
```

**Review by david-risney — CHANGES_REQUESTED** [review]

File: specs/Media.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                return S_OK;
+            })
+            .Get(),
+        &m_isAudioPlayingChangedToken));
+}
+
+// Mute the current window and show a mute icon on the title bar
+ void AudioComponent::Mute()
+ {
+    CHECK_FAILURE(m_webView->Mute());
```
[the remark this answers is not in the document]
Comment (dianaqu) [reply specs/Media.md:59]: Updated

File: specs/Media.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2IsAudioPlayingChangedEventHandler;
+
+[uuid(71c906d9-4a4d-4dbe-aa1b-db64f4de594e), object, pointer_default(unique)]
+interface ICoreWebView2_6 : ICoreWebView2 {
```
[the remark this answers is not in the document]
Comment (dianaqu) [reply specs/Media.md:118]: Updated

File: specs/Media.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+{
+
+    runtimeclass CoreWebView2
+    {
+        // ...
+            Boolean IsMuted { get; };
+            Boolean IsAudioPlaying { get; };
+
+            event Windows.Foundation.TypedEventHandler<CoreWebView2, Object> IsAudioPlayingChanged;
+            void Mute();
```
[the remark this answers is not in the document]
Comment (dianaqu) [reply specs/Media.md]: Making it settable. End developer doesn't seem to have a preference and I update the sample code

**Comment by Symbai** [comment]
It would be nice if the developer gets the source of the media which requested audio. For example if a developer wants to allow media of a specific source but disallow media from everywhere else, its kinda hard/impossible to achieve this with this API.

**Comment by dianaqu** [comment]
> It would be nice if the developer gets the source of the media which requested audio. For example if a developer wants to allow media of a specific source but disallow media from everywhere else, its kinda hard/impossible to achieve this with this API.

Would you mind elaborate on what you mean? This current spec is only to get/set the mute state or get the audible state. We don't hold on any sources

**Comment by Symbai** [comment]
> 
> Would you mind elaborate on what you mean? This current spec is only to get/set the mute state or get the audible state. We don't hold on any sources

I want to (un)mute audio when it comes from a specific URL (the media, not the website), rather than mute/unmute whole tab. Unless that is not possible in Edge itself? In this case nevermind please.

**Comment by oldnewthing** [comment]
> I want to (un)mute audio when it comes from a specific URL (the media, not the website), rather than mute/unmute whole tab. Unless that is not possible in Edge itself? In this case nevermind please.

This API does not provide that degree of control. This API controls muting at the WebView level.

It gets complicated if there is more than one media element on the page, or if that media element has multiple sources. I guess you can inject script into the page that hunts down all `<video>` elements and then studies each one to decide whether to mute it. (And use a MutationObserver to learn when new `<video>` elements are added.)

**Pull request merged**